### ISSUE TITLE
Error when applying castNonNull() to a @MonotonicNonNull type

### ISSUE THREAD

**Issue by trask** [typetools/checker-framework#1555]
E.g.

```
import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
import org.checkerframework.checker.nullness.NullnessUtil;

class Test {

    @MonotonicNonNull
    private String x;

    String test() {
        return NullnessUtil.castNonNull(x);
    }
}
```

Results in the following error:

```
Test.java:10: error: [monotonic.type.incompatible] cannot assign org.checkerframework.checker.nullness.qual.MonotonicNonNull to org.checkerframework.checker.nullness.qual.MonotonicNonNull (monotonic type).
        return NullnessUtil.castNonNull(x);
                                        ^
  full type found: @Initialized @MonotonicNonNull String
1 error
```

**Comment by smillst** [comment]
Thanks for reporting this.  This is a false positive related to issue #1394.  However, we should be able to work around this particular case by annotating the parameter of `castNonNull` with `@Nullable`.

**Comment by mernst** [comment]
Right now the signature of `castNonNull` is

```java
<T extends /*@Nullable*/ Object> /*@NonNull*/ T castNonNull(T ref)
```

Are you proposing to change it to

```java
<T extends Object> T castNonNull(/*@Nullable*/ T ref)
```
or to something else?

**Comment by smillst** [comment]
```java
<T extends /*@Nullable*/ Object> /*@NonNull*/ T castNonNull(/*@Nullable*/ T ref)
```
